include/stack_macros.h

```diff
@@ -45,13 +45,21 @@
 
 /*-----------------------------------------------------------*/
 
+/*
+ * portSTACK_LIMIT_PADDING is a number of extra words to consider to be in
+ * use on the stack.
+ */
+#ifndef portSTACK_LIMIT_PADDING
+    #define portSTACK_LIMIT_PADDING 0
+#endif
+
 #if ( ( configCHECK_FOR_STACK_OVERFLOW == 1 ) && ( portSTACK_GROWTH < 0 ) )
 
 /* Only the current stack state is to be checked. */
     #define taskCHECK_FOR_STACK_OVERFLOW()                                                            \
     {                                                                                                 \
         /* Is the currently saved stack pointer within the stack limit? */                            \
-        if( pxCurrentTCB->pxTopOfStack <= pxCurrentTCB->pxStack )                                     \
+        if( pxCurrentTCB->pxTopOfStack <= pxCurrentTCB->pxStack + portSTACK_LIMIT_PADDING)            \

```

Comment:
The PendSV handler does not save the SP with extra words 4 in TCB - so this check would not do detect the boundary overflow you are trying to detect? Therefore, I suggest that we remove this common change.

Reply:
can you clarify what you mean?

Reply:
i mean the point is that the value saved does not include the extra words, which is why it is checked here... if you save a different pointer in the TCB then it breaks the ability to debug with openocd etc.

Note that this is the only piece of code I found that actually cared whether there was something below `pxTopOfStack`